wallet/src/actors/worker/methods.rs

```diff
@@ -749,9 +752,14 @@ impl Worker {
             )]);
             self.notify_client(&wallet, sink.clone(), events).ok();
 
+            // If the batch size smaller than the limit, we reached the end. Set syncing to false
+            if batch_size < i128::from(limit) {
+                wallet.unset_syncing()?;
```

Comment:
If there is any error during synchronization, this will never be called. To avoid any problems, we can set and unset the syncing flag in the `sync` function instead of here. This is why we have a `sync_inner` function.

Reply:
lets do that